Src/Newtonsoft.Json/JsonWriter.cs

```diff
@@ -380,10 +369,10 @@ private JsonContainerType Pop()
         {
             JsonPosition oldPosition = _currentPosition;
 
-            if (_stack != null && _stack.Count > 0)
+            if (_stack?.Count > 0)
             {
-                _currentPosition = _stack[_stack.Count - 1];
-                _stack.RemoveAt(_stack.Count - 1);
+                _currentPosition = _stack.Last();
+                _stack.Remove(_stack.Last());
```

Comment:
` _stack.RemoveAt(_stack.Count - 1);` is not the same as  `_stack.Remove(_stack.Last());`

Reply:
Cancelled this too